packages/list/index.tsx

```diff
@@ -185,16 +200,30 @@ export default class List<T extends HTMLElement = HTMLElement> extends React.Com
           listItem.setAttribute(attr, value);
         }
       },
+      /**
+       * Pushes class name to state.listItemClassNames[listItemIndex] if it doesn't yet exist.
+       */
       addClassForElementIndex: (index, className) => {
-        const listItem = this.listElements[index];
-        if (listItem) {
-          listItem.classList.add(className);
+        const {listItemClassNames} = this.state;
+        if (listItemClassNames[index] && listItemClassNames[index].indexOf(className) === -1) {
+          listItemClassNames[index].push(className);
+        } else {
+          listItemClassNames[index] = [className];
         }
+        this.setState({listItemClassNames});
```

Comment:
Did you test this with disabled list item? 

`React.Children.map` runs on each list item but these adapters excludes disabled list item when indexing.

Reply:
good catch - updated and added tests